spark-mongodb/src/main/scala/com/stratio/datasource/mongodb/config/MongodbConfig.scala

```diff
@@ -130,16 +134,25 @@ object MongodbConfig {
   }
 
   /**
+   * Parse one key to the associated readPreferenceTags
+   * @param readPreferenceTags string key for identify the correct object
+   * @return TagSet object
+   */
+  def parseReadPreferenceTags(readPreferenceTags: String): TagSet = {
+    return new TagSet( readPreferenceTags.split(",").map(_.split(":")).filter(_.length == 2).map( kv => new Tag(kv(0),kv(1)) ).toList )
+  }
+
+  /**
    * Parse one key to the associated readPreference
    * @param readPreference string key for identify the correct object
    * @return readPreference object
    */
-  def parseReadPreference(readPreference: String): ReadPreference = {
+  def parseReadPreference(readPreference: String, readPreferenceTags: String): ReadPreference = {
     readPreference.toUpperCase match {
       case "PRIMARY" => com.mongodb.casbah.ReadPreference.Primary
-      case "SECONDARY" => com.mongodb.casbah.ReadPreference.Secondary
-      case "NEAREST" => com.mongodb.casbah.ReadPreference.Nearest
-      case "PRIMARYPREFERRED" => com.mongodb.casbah.ReadPreference.primaryPreferred
+      case "SECONDARY" => com.mongodb.casbah.ReadPreference.secondary(parseReadPreferenceTags(readPreferenceTags))
```

Comment:
Could you import these values so you don't have to repeat its package path 5 times?
Something like `import com.mongodb.casbah.ReadPreference.{Primary, Secondary, SecondaryPreferred, Nearest}`